### PR TITLE
Adds Dnn.ClientDependency to Platform to UNBLOCK local development

### PR DESCRIPTION
**Pull request by ahoefling**
Fixes broken development branch by removing the git submodule and replacing it with the appropriate nuget reference.

-  Updated Dnn.ClientDependency to be published to the Dnn MyGet v1.9.2.7
-  Added NuGet feeds to the NuGet.Config file. This enables automatic nuget feed reference so you don't need to manually update your references
    * `https://api.nuget.org/v3/index.json`
    * `https://www.myget.org/F/dnn-software-public/api/v2`
- Removed git submodule and project sln reference to `ClientDependency.Core`
- Added NuGet reference to `Dnn.ClientDependency` to all projects that require it

Related: dnnsoftware/ClientDependency#3 dnnsoftware/ClientDependency#6
Fixes: #2061

### CODE REVIEW

**Comment by tpluscode** [comment]
LGTM

**Comment by galatrash** [comment]
Guys, this PR has broken the CI builds because you removed Client Dependency as submodule, but left the .gitmodules file there. You should have done 'git deinit -f --all' to remove the submodule first then delete this file. I've took care of this anyway.

**Comment by galatrash** [comment]
Moreover, there are still 3 projects which still reference the removed 'ClientDependency.Core.csproj'. These are:
* Dnn.Platform\DNN Platform\Modules\HtmlEditorManager\DotNetNuke.Modules.HtmlEditorManager.csproj
* Dnn.Platform\DNN Platform\Modules\RazorHost\DotNetNuke.Modules.RazorHost.csproj
* Dnn.Platform\Website\DotNetNuke.Website.csproj

Please fix these.

**Comment by mitchelsellers** [comment]
@galatrash & others.  How does the community have the ability to validate functionality inside of the current CI Build Process?  Can we improve the pull request process to prevent this from happening in the future?

File: DNN_Platform.sln [hunk 1]
```diff
@@ -95,7 +95,7 @@ Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "Integration", "Integration"
 EndProject
 Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "DotNetNuke.Tests.Urls", "DNN Platform\Tests\DotNetNuke.Tests.Urls\DotNetNuke.Tests.Urls.csproj", "{F9CE7C63-C729-4E3A-A266-6B23D75A1247}"
 EndProject
-Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "DotNetNuke.Tests.AspNetCCP", "DNN Platform\Tests\DotNetNuke.Tests.AspNetCPP\DotNetNuke.Tests.AspNetCPP.csproj", "{64BA74E9-411E-4DC1-91CB-DEF7F22468AD}"
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "DotNetNuke.Tests.AspNetCPP", "DNN Platform\Tests\DotNetNuke.Tests.AspNetCPP\DotNetNuke.Tests.AspNetCPP.csproj", "{64BA74E9-411E-4DC1-91CB-DEF7F22468AD}"
```
[the remark this answers is not in the document]
Comment (ahoefling) [reply DNN_Platform.sln]: I didn't explicitly make this change visual studio must have made the change. When I get back to looking at this I will take a look at this

**Comment by ahoefling** [comment]
I was expecting a little bit more discussion on this PR since it was such a large change to the platform that could have serious side-effects. That set aside, AFAIK there is no github integration for any type of CI build which makes it difficult to know if there is any problem.

I know the Technology Advisory Group has been talking about CI but if there is an existing CI that is currently used it needs to be integrated into GitHub as a validation check so we know if the Pull Request is safe to merge or not.

**Comment by tpluscode** [comment]
@mitchelsellers, let's start by re-enabling AppVeyor CI on Pull Requests, which I did, and adding the configuration file to the repository (#2067).

I also created another, to get rid of all remaining ClientDependency.Core references (#2068), and fixed that abbreviated project name (#2069).

**Comment by mitchelsellers** [comment]
@tpluscode Perfect!  Looks like the builds are failing though based on all of those pull requests?

**Comment by tpluscode** [comment]
Indeed. There are 4 failing tests. Two coming from ClientDependency and two from somewhere else

**Comment by galatrash** [comment]
@mitchelsellers we are working on adding PR builds into CI so they can be validated before being merged.

**Pull request merged**